File: spring-cloud-integrated-demo/account-service/account-service-api/src/main/java/com.github.jccode.springcloud.integrateddemo.account/api/AccountAPI.java
package com.github.jccode.springcloud.integrateddemo.account.api;


import com.github.jccode.springbootsample.core.data.rest.RestResult;
import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

@FeignClient(value = "account-service", fallback = AccountAPIFallback.class)
public interface AccountAPI {

    @RequestMapping(value = "/users/{id}/balance", method = RequestMethod.GET)
    RestResult<Integer> getBalance(@PathVariable("id") Integer userId);

}

File: spring-cloud-integrated-demo/account-service/account-service-api/src/main/scala/com/github/jccode/springcloud/integrateddemo/account/api/AccountApi.scala
package com.github.jccode.springcloud.integrateddemo.account.api

import com.github.jccode.springbootsample.core.data.RestResult.RestSuccess
import org.springframework.cloud.netflix.feign.FeignClient
import org.springframework.web.bind.annotation.{PathVariable, RequestMapping, RequestMethod}


@FeignClient(name = "account-service", fallback = classOf[AccountApiFallback])
trait AccountApi {

  @RequestMapping(value = Array("/users/{id}/balance"), method = Array(RequestMethod.GET))
  def getBalance(@PathVariable("id") userId: Integer): RestSuccess[Integer]

}
